pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -704,10 +705,23 @@ public Optional<DispatchRateLimiter> getRateLimiter() {
     }
 
     @Override
-    public void initializeDispatchRateLimiterIfNeeded() {
-        if (!dispatchRateLimiter.isPresent()
-            && DispatchRateLimiter.isDispatchRateEnabled(topic.getReplicatorDispatchRate())) {
-            this.dispatchRateLimiter = Optional.of(new DispatchRateLimiter(topic, Type.REPLICATOR));
+    public boolean initializeDispatchRateLimiterIfNeeded() {
+        synchronized (dispatchRateLimiterLock) {
+            if (!dispatchRateLimiter.isPresent()
+                && DispatchRateLimiter.isDispatchRateEnabled(topic.getReplicatorDispatchRate())) {
+                this.dispatchRateLimiter = Optional.of(new DispatchRateLimiter(topic, Type.REPLICATOR));
+                return true;
+            }
+            return false;
+        }
+    }
+
+    @Override
+    public void updateRateLimiter() {
+        synchronized (dispatchRateLimiterLock) {
```

Comment:
Is this necessary? `initializeDispatchRateLimiterIfNeeded` already has synchronized `dispatchRateLimiterLock`.

Reply:
@Jason918  It seems not necessary. I will remove the `synchronized ` here.  Because `updateDispatchRate(dispatchRate)`, which is located at Line203 is synchronized:

https://github.com/apache/pulsar/blob/d4be6bb05204c270122e159c5c361cd7a2412805/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/DispatchRateLimiter.java#L178-L204